airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/errors/messages/ErrorMessage.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.base.errors.messages;
+
+import java.util.Objects;
+
+public class ErrorMessage {
+
+  public static String getDefaultErrorMessage(String stateCode, Exception exception) {
+    return configMessage(stateCode, 0, exception.getLocalizedMessage());
+  }
+
+  public static String getErrorMessage(String stateCode, int errorCode, String message, Exception exception) {
+    if (Objects.isNull(message)) {
+      return getDefaultErrorMessage(stateCode, exception);
+    } else {
+      return configMessage(stateCode, errorCode, message);
+    }
+  }
+
+  private static String configMessage(String stateCode, int errorCode, String message) {
+    var stateCodePart = Objects.isNull(stateCode) ? "" : "State code: " + stateCode + "; ";
+    var errorCodePart = errorCode == 0 ? "" : "Error code: " + errorCode + "; ";
+    return stateCodePart + errorCodePart + "Message: " + message;
```

Comment:
Can we potentially use a `String.format()` here? How come the code is using type inference with `var` since it's clear the string will always be of type string , no?

Reply:
Changed the generation of `errorMessage` using `String.format()`. And using `var` is my habit. Replaced it with String